class Solution {
    public boolean validPath(int n, int[][] edges, int source, int destination) {
        ArrayList<ArrayList<Integer>>adj=new ArrayList<>();
        for(int i=0;i<n;i++){
            adj.add(new ArrayList<>());
        }
        for(int edge[]:edges){
            adj.get(edge[0]).add(edge[1]);
            adj.get(edge[1]).add(edge[0]);
        }
        boolean []visited=new boolean[n];
        return dfs(source,destination,adj,visited);
    }
    static boolean dfs(int node,int destination,ArrayList<ArrayList<Integer>>adj,boolean[]visited){
        if(node==destination)return true;
        visited[node]=true;
        for(int neighbor:adj.get(node)){
            if(!visited[neighbor] && dfs(neighbor,destination,adj,visited)){
                  return true;
                }
        }
        return false;


    }
}